tests/baselines/reference/excessPropertyCheckIntersectionWithRecursiveType.errors.txt

```diff
@@ -81,4 +84,65 @@ tests/cases/compiler/excessPropertyCheckIntersectionWithRecursiveType.ts(52,9):
         },
       },
     }
+    
+    // repro from #40405
+    
+    type Length<T extends any[]> = T["length"];
+    type Prepend<V, T extends any[]> = ((head: V, ...args: T) => void) extends (
+      ...args: infer R
+    ) => void
+      ? R
+      : any;
+    
+    type BuildTree<T, N extends number = -1, I extends any[] = []> = {
+      1: T;
+      0: T & { children: BuildTree<T, N, Prepend<any, I>>[] };
+    }[Length<I> extends N ? 1 : 0];
+    
+    interface User {
+      name: string;
+    }
+    
+    type GrandUser = BuildTree<User, 2>;
+    
+    const grandUser: GrandUser = {
+      name: "Grand User",
+      children: [
+        {
+          name: "Son",
+          children: [
+            {
+              name: "Grand son",
+              children: [

```

Comment:
Unrelated but your suggestion to move diagnostics around would make this less... squiggly, right?

Reply:
Hm, yes - but that wasn't covered by more PR. I only touched `return` and `yield` keywords. This one would have to "decorate" the property's key instead of the whole property. On principle though - yes.